### PR TITLE
add: Migration from Caching to Valkey

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
## Describe your changes

Added migration process from Aiven for Caching or Valkey to Aiven for Valkey.

[MA-3046](https://aiven.atlassian.net/browse/MA-3046)

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](styleguide.md).
- [x] My links start with `/docs/`.

[MA-3046]: https://aiven.atlassian.net/browse/MA-3046?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+If the migration fails, resolve the issue and click **Start over**.
+
+### Step 5: Close and complete the migration
+
+After the migration:
+
+- **Stop replication**: If no further synchronization is needed, and you are ready
+  to switch to Aiven for Valkey after testing.
+- **Keep replicating**:If continuous data synchronization is needed. Avoid system
```
Comment (bogdanp05) [inline docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md:85]: there should be a space between `:` and `If`

**Review by bogdanp05 — CHANGES_REQUESTED** [review]
looks good apart from the typo

File: docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md (hunk 1 above)
Comment (harshini-rangaswamy) [reply docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md:85]: fixed

**Review by bogdanp05 — APPROVED** [review]
lgtm

File: docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+- A target [Aiven for Valkey service](/docs/products/valkey/get-started).
+- Source database details:
+  - **Hostname or connection string**: The public hostname, connection string, or
+    IP address used to connect to the database.
+  - **Port**: The port used to connect to the database.
+  - **Username**: The username with sufficient permissions to access the data.
+  - **Password**: The password used to connect to the database.
+- Firewall rules updated or temporarily disabled to allow traffic between source and
+  target databases.
+- A source Caching or Valkey service secured with SSL.
```
Comment (wojcik-dorota) [inline docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md:22]:
```suggestion
- A source Aiven for Caching or Valkey service secured with SSL.
```
And if this comment is valid, I'd fix all the names in the article accordingly.

File: docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+1. Log in to the [Aiven Console](https://console.aiven.io/) and select the
+   Aiven for Valkey service for your migration.
+1. Go to <ConsoleLabel name="service settings"/> from the sidebar.
+1. Scroll to <ConsoleLabel name="actions"/> > **Migrate database** to start the
+   migration.
+1. Follow the wizard to guide you through the migration process.
+
+### Step 1: Configure
+
+Review the prerequisites and click **Get started** to begin.
```
Comment (wojcik-dorota) [inline docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md:37]: My 1st thought was to get back to the previous section in the article to `review the prerequisites` so maybe it'd be good to indicate where to perform the actions? e.g.

```suggestion
In the migration wizard, review the prerequisites and click **Get started** to begin.
```

File: docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+- Password
+
+Select **SSL encryption recommended** for a secure connection, and click **Run check**.
+
+The [Aiven Console](https://console.aiven.io/) validates the database configurations. If
+any errors occur, follow the on-screen instructions to resolve them and rerun the check.
+
+:::note
+The migration does not include service user accounts or commands in progress.
+:::
```
Comment (wojcik-dorota) [inline docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md:55]: How about moving this note upwards, somewhere around `Prerequisites`, before the users start the migration process?

File: docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+### Step 3: Migrate
+
+Once validation is complete, click **Start migration** to begin migrating data to
```
Comment (wojcik-dorota) [inline docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md:59]:
```suggestion
Once the validation is complete, click **Start migration** in the wizard to begin migrating data to
```

File: docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+- Do not create or delete databases on the source service.
+- Avoid network or configuration changes that might disrupt the connection between source
+  and target databases, such as firewall modifications.
+
+If the migration fails, resolve the issue and click **Start over**.
+
+### Step 5: Close and complete the migration
+
+After the migration:
```
Comment (wojcik-dorota) [inline docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md:81]: Are the two bold options below actual buttons? If so

```suggestion
After the migration, select one of the following:
```

**Review by wojcik-dorota — APPROVED** [review]
A few tiny suggestions

**Pull request merged**